### ISSUE TITLE
Feature Request: Audio CD Support

### ISSUE THREAD

**Issue by lostgamer-ops13** [flyinghead/flycast#1654]
Hi there

I would really like if you added Audio CD support to the emulator please. I used to use this extensively on my OG Dreamcast console but unfortunately due to age, the laser is wearing down. I understand this might be a niche feature but for nostalgic purposes, I would really appreciate this very much. I have over 1000+ CDs that I would like to test out again 😊🙂

Thanks!!!

![Screenshot (74)](https://github.com/user-attachments/assets/b70e5dc8-6b29-47b5-a333-a5de03f47e5c)

**Comment by flyinghead** [comment]
Feature available in the latest `dev` build.
Please test and report.

**Comment by kihato** [comment]
On Windows, you will need to prepare libcdio-19.dll and libiconv-2.dll beforehand.

We tested by inserting an actual MIL-CD into the drive.
There are absolutely no problems with playback of audio tracks or the extra parts.
However, in the case of MIL-CD, perhaps because of the extra tracks, the animation of the disc spinning does not seem to stop even after the last music track has finished playing.
When playing an Audio CD, the disc rotation animation stops once the final track has been played without any problems.

https://github.com/user-attachments/assets/4fba0d97-f6b0-42f1-a7ae-60d145fcbe8e

**Comment by flyinghead** [comment]
> On Windows, you will need to prepare libcdio-19.dll and libiconv-2.dll beforehand.

Right, I forgot about this. I will try to statically link instead.

> However, in the case of MIL-CD, perhaps because of the extra tracks(...)

With multisession CDs (CDROM XA), there is a 02:32 gap between session 1 (audio tracks) and 2 (data track) that corresponds to the session lead out then lead in. So it's possible that this gap is being played? Or perhaps an error should be returned at this point?

**Comment by kihato** [comment]
You're right, after waiting 2 minutes 32 seconds the animation stopped!
The logs showed sector read miss messages continuing to be logged for 2 minutes and 32 seconds.

**Comment by flyinghead** [comment]
A quick test on a real Dreamcast reveals that the track stops playing and the disc stops spinning at the end of the track. So something isn't right with flycast. Note that the same happens with CDIs and multisession CUEs.

**Comment by flyinghead** [comment]
Fixed the libcdio-19.dll and libiconv-2.dll requirement.
Audio playback now stops when reaching the session lead-out.
(`dev` branch)

**Comment by kihato** [comment]
I have confirmed that the last audio track stops properly on MIL-CD. 
Single track repeat and all track repeat also work fine.
Thanks for your great work!
